### PR TITLE
feat(backend): add concrete-frame skill core implementation (Part 4/4)

### LINKED ISSUE #238: feat(skills): add concrete frame, shear wall, and additional structural design skills / 补充混凝土框架、剪力墙等结构设计技能
## Summary

Currently only the steel frame skill exists under `backend/src/agent-skills/structure-type/`. Additional structural design skills are needed to cover common building types: concrete frames, shear walls, steel-concrete composite structures, and other standard structural systems.

## Current State

Existing skills in `backend/src/agent-skills/structure-type/`:
- Steel frame (frame skill) — covers portal frames, multi-story steel frames

Missing skills:
- Concrete frame (reinforced concrete beams, columns, slabs)
- Shear wall (wall-frame systems, core walls)
- Steel-concrete composite (composite beams, columns)
- Other common types: truss (already partially in analysis), foundation, floor slab

### Concrete Frame Skill

- [x] `skill.yaml` with `id: concrete-frame`, triggers: `混凝土框架`, `concrete frame`, `RC frame`
- [x] `intent.md` / `draft.md` / `analysis.md` / `design.md` stage markdowns
- [x] `handler.ts` implementing `SkillHandler`:
  - Concrete material grade parsing (C20–C80)
  - Rectangular/T-shaped beam sections, rectangular/circular column sections
  - Slab span and thickness estimation
  - Code-check against GB 50010 (concrete design)
- [x] DraftState extensions for concrete-specific fields
- [x] Tests in `__tests__/`

### Shear Wall Skill

- [ ] `skill.yaml` with `id: shear-wall`, triggers: `剪力墙`, `shear wall`, `wall-frame`
- [ ] Stage markdowns for wall layout, thickness, reinforcement
- [ ] `handler.ts`:
  - Wall segment geometry (length, thickness, openings)
  - Coupling beam design
  - Seismic grade and ductility requirements (GB 50011)
- [ ] Tests in `__tests__/`

### Steel-Concrete Composite Skill

- [ ] `skill.yaml` with `id: composite`, triggers: `组合结构`, `composite`, `steel-concrete`
- [ ] Stage markdowns for composite beam/column design
- [ ] `handler.ts`:
  - Steel profile + concrete slab interaction
  - Shear stud design
  - Code-check against GB 50017 composite sections
- [ ] Tests in `__tests__/`

### Foundation & Slab Skills (lower priority)

- [ ] Independent/pile foundation skill
- [ ] Floor slab design skill (one-way, two-way, flat slab)

## Technical Considerations

| Concern | Approach |
|---------|----------|
| Skill interface | Follow existing `SkillHandler` pattern from frame skill |
| Code-check | Reuse existing `code-check` skill runners (GB 50010, GB 50011, GB 50017) |
| Material | Extend existing material definitions in `agent-skills/material/` |
| DraftState | Extend shared `DraftState` type with concrete/wall-specific fields |
| Priority | Concrete frame first (most common), then shear wall, then composite |

## Dependencies

- Existing frame skill as reference implementation
- Material definitions in `agent-skills/material/`
- Code-check runners for GB 50010, GB 50011, GB 50017

## Done When

- [ ] Concrete frame skill with full handler and tests
- [ ] Shear wall skill with full handler and tests
- [ ] Steel-concrete composite skill with handler and tests
- [ ] All skills auto-discovered and routable via Agent

---

## 概述

当前 `backend/src/agent-skills/structure-type/` 中仅有钢框架技能。需要补充更多结构设计技能以覆盖常见建筑类型：混凝土框架、剪力墙、钢-混凝土组合结构及其他标准结构体系。

## 现状

`backend/src/agent-skills/structure-type/` 中已有技能：
- 钢框架（frame skill）— 覆盖门式刚架、多层钢框架

缺失技能：
- 混凝土框架（钢筋混凝土梁、柱、板）
- 剪力墙（框-剪体系、核心筒）
- 钢-混凝土组合结构（组合梁、组合柱）
- 其他常见类型：桁架（分析部分已有）、基础、楼板

### 混凝土框架技能

- [x] `skill.yaml`，`id: concrete-frame`，触发词：`混凝土框架`、`concrete frame`、`RC frame`
- [x] `intent.md` / `draft.md` / `analysis.md` / `design.md` 阶段 markdown
- [x] `handler.ts` 实现 `SkillHandler`：
  - 混凝土材料等级解析（C20–C80）
  - 矩形/T 形梁截面、矩形/圆形柱截面
  - 板跨和板厚估算
  - GB 50010（混凝土设计）规范验算
- [x] DraftState 扩展（混凝土专用字段）
- [x] `__tests__/` 测试

### 剪力墙技能

- [ ] `skill.yaml`，`id: shear-wall`，触发词：`剪力墙`、`shear wall`、`wall-frame`
- [ ] 墙体布置、厚度、配筋阶段 markdown
- [ ] `handler.ts`：
  - 墙段几何（长度、厚度、洞口）
  - 连梁设计
  - 抗震等级和延性要求（GB 50011）
- [ ] `__tests__/` 测试

### 钢-混凝土组合结构技能

- [ ] `skill.yaml`，`id: composite`，触发词：`组合结构`、`composite`、`steel-concrete`
- [ ] 组合梁/柱设计阶段 markdown
- [ ] `handler.ts`：
  - 钢型材 + 混凝土板相互作用
  - 栓钉设计
  - GB 50017 组合截面规范验算
- [ ] `__tests__/` 测试

### 基础与楼板技能（低优先级）

- [ ] 独立/桩基础技能
- [ ] 楼板设计技能（单向板、双向板、无梁板）

## 技术考虑

| 考虑项 | 方案 |
|--------|------|
| 技能接口 | 遵循 frame skill 的 `SkillHandler` 模式 |
| 规范验算 | 复用 `code-check` 技能运行器（GB 50010、GB 50011、GB 50017） |
| 材料 | 扩展 `agent-skills/material/` 中现有材料定义 |
| DraftState | 用混凝土/剪力墙专用字段扩展共享 `DraftState` 类型 |
| 优先级 | 先混凝土框架（最常见），再剪力墙，再组合结构 |

## 依赖

- 现有 frame skill 作为参考实现
- `agent-skills/material/` 材料定义
- GB 50010、GB 50011、GB 50017 规范验算运行器

## 完成标准

- [ ] 混凝土框架技能包含完整 handler 和测试
- [ ] 剪力墙技能包含完整 handler 和测试
- [ ] 钢-混凝土组合结构技能包含 handler 和测试
- [ ] 所有技能自动发现且可通过 Agent 路由

## 相关文件

- `backend/src/agent-skills/structure-type/frame/` — 参考实现
- `backend/src/agent-runtime/types.ts` — SkillHandler 接口
- `backend/src/agent-skills/code-check/` — 规范验算运行器
- `backend/src/agent-skills/material/` — 材料定义

### PR DESCRIPTION
**Pull request by LaobaiAi**
Related to: #238 (concrete frame skill — Part 4  of 4       final PR)

----------------------------------------
## What's In This PR

Completes the concrete-frame skill by filling test coverage gaps, populating LLM test cases, and cleaning all development markers. This is the final PR for the concrete-frame skill.

Base specification: GB/T 50010-2010 §9.2.1 / §9.3.1

### Changed

| Layer                | What changed                                                                 | Why                                    |
|----------------------|------------------------------------------------------------------------------|----------------------------------------|
| `model.ts`           | Add `bar_count`, `sn` (net spacing) to beam/column rebar metadata            | Enable clear spacing code-check        |
| `entry.ts`           | Pass `bar_count`, `sn` through elementData bridge → Python                   | Complete rebar metadata pipeline       |
| `code_check.py`      | Add `钢筋净距` check + `构造验算` group for beams and columns               | GB50010 §9.2.1/§9.3.1 detailing        |
| `test_gb50010.py`    | 6 net spacing tests + fix 3 indices for new check group structure            | Coverage for pass/violation/skip       |
| `skill-core.test.mjs`| 13 design-conditions unit tests + 3 3D model error path tests                | Edge case coverage for all 4 helpers   |
| `handler.test.mjs`   | 3 e2e tests verifying bar_count/sn metadata on 2D/3D model elements          | Full metadata pipeline validation      |
| `cases.json`         | 9 LLM test scenarios (routing zh/en, extraction, irregular rejection)        | Auto-discovery test completeness       |
| `handler.ts` etc.    | Clean PR2/PR3/M4 comments from 7 files                                       | Final polish for complete skill        |

---

## Checks Now Computed With Real Values (Complete Set)

| Check (Chinese)       | Clause       | Data consumed                                     | Added In |
|------------------------|-------------|---------------------------------------------------|----------|
| 轴压比                 | §6.2.15      | N / (fc·A)                                        | PR3      |
| 长细比                 | §6.2.20      | l₀/i / 120                                        | PR3      |
| 偏心受压               | §6.2.17      | N-Nu + M-Mu interaction with φ stability          | PR3      |
| 正截面受弯             | §6.2.1/10    | α1·fc·b·x·(h0−x/2) with model As                 | PR3      |
| 斜截面受剪             | §6.3.1/12    | Vc + Vs with model Asv, spacing                   | PR3      |
| 挠度                  | §3.3.2       | 5Ml²/(48·0.85Ec·I) / (l/250)                     | PR3      |
| 裂缝宽度              | §7.1.2       | αcr·ψ·σsk/Es·(1.9c+0.08deq/ρte) / 0.3            | PR3      |
| ✨ 钢筋净距            | §9.2.1/9.3.1 | sₙ / max(1.5d, {30\|50}) — **this PR**           | PR4      |

All 8 checks: **0 mock**, real OpenSees forces + model design values.

---

## Impacted Areas

- `backend/agent-skills/structure-type/concrete-frame/`
- `backend/agent-skills/code-check/gb50010/`
- `backend/agent-skills/code-check/entry.ts`
- No frontend, scripts, or other skill changes

---

## Commands Run and Results

```
npm run build --prefix backend
# ✓ TypeScript compilation successful

npm test --prefix backend -- --runInBand
# ✓ 92 suites, 983 tests passing

python -m pytest gb50010/__tests__/test_gb50010.py -v
# ✓ 39 passed
```

---

## What's Not In This PR

- 净距验算使用保守上界 (梁: max(1.5d,30), 柱: max(1.5d,50))，未区分梁顶/底筋差异
- 未新增分析引擎 (PKPM/YJK) 回归 — 已在 PR3 中覆盖

---

## Next Steps

This is the final PR for the concrete-frame skill per #238. After merge, the skill is complete.

### CODE REVIEW

File: backend/src/agent-skills/code-check/gb50010/code_check.py [hunk 1]
```diff
@@ -434,6 +434,22 @@ def _has_override(key: str) -> bool:
...
+    main_dia = elem.get('main_dia')
+    if isinstance(bar_count, (int, float)) and isinstance(sn, (int, float)) and isinstance(main_dia, (int, float)):
+        try:
+            elem_type = _resolve_element_type(elem_id, context)
+            if elem_type == 'column':
+                sn_limit = max(1.5 * float(main_dia), 50.0)  # §9.3.1
+            else:
+                sn_limit = max(1.5 * float(main_dia), 30.0)  # §9.2.1 (top bars, conservative)
+            if sn_limit > 0 and float(sn) > 0:
+                computed['钢筋净距'] = sn_limit / float(sn)
```
[the remark this answers is not in the document]
Comment (LaobaiAi) [reply backend/src/agent-skills/code-check/gb50010/code_check.py]: It has been modified as suggested

File: backend/src/agent-skills/structure-type/concrete-frame/model.ts [hunk 2]
```diff
@@ -1085,12 +1094,24 @@ export function buildConcreteFrameModel(
   const colH = columnProps.height ?? 500;
   const colRhoMin = Math.max(0.006, 0.55 * concreteProps.ft / rebarProps.fy);
   const colAs = Math.max(colB * colH * colRhoMin, Math.PI * 20 * 20);
+  const colMainDia = 20;
+  const colBarArea = Math.PI * colMainDia * colMainDia / 4;
+  const colBarCount = Math.max(4, Math.ceil(colAs / colBarArea));
+  // GB50010-2010 §9.3.1: 净距 sₙ per side face
+  const colBarCountPerSide = Math.max(2, Math.ceil(colBarCount / 4));
```
[the remark this answers is not in the document]
Comment (LaobaiAi) [reply backend/src/agent-skills/structure-type/concrete-frame/model.ts]: It has been modified as suggested

File: backend/src/agent-skills/code-check/gb50010/code_check.py [hunk 3]
```diff
@@ -473,6 +492,13 @@ def _check_beam(checker: Any, elem_id: str, context: Dict[str, Any]) -> List[Dic
                 checker._calc_item(elem_id, '斜截面受剪', context, 'GB50010-2010 6.3.1', 'V <= Vc + Vs', 0.95),
             ],
         },
+        {
+            'chapter': '第9章 构造规定',
+            'name': '构造验算',
+            'items': [
+                checker._calc_item(elem_id, '钢筋净距', context, 'GB50010-2010 9.2.1', 's_n >= max(1.5d, 30)', 1.0),
```
Comment (qinsz01) [inline backend/src/agent-skills/code-check/gb50010/code_check.py:514]: This looks inconsistent with the intended "skip when spacing metadata is missing" behavior. `_compute_utilization_overrides()` deliberately leaves `钢筋净距` unset when `bar_count` / `sn` / `main_dia` are incomplete, but `_check_beam()` always emits the `钢筋净距` item. Because `CodeChecker._calc_item()` falls back to a deterministic utilization when no override exists, a beam without spacing metadata will still show a passing net-spacing check instead of being skipped or marked not applicable. The same issue applies to the column item below. Could we gate these items on computed spacing availability, or otherwise teach this check to return an explicit skipped/not-applicable result, and add a full `check_element()` test for the missing-metadata case?

**Comment by qinsz01** [comment]
Additional review note: `backend/src/agent-skills/code-check/gb50010/code_check.py` `get_rules()` still advertises the previous beam/column check lists and omits the new `钢筋净距` check that `check_element()` now performs. If this metadata feeds capability, report, or traceability display, consumers will under-report the actual checks being run. Please add `钢筋净距` to both the beam and column `checks` arrays, and cover that in the rule metadata test if one exists.

File: backend/src/agent-skills/code-check/gb50010/code_check.py (hunk 3 above)
Comment (LaobaiAi) [reply backend/src/agent-skills/code-check/gb50010/code_check.py:514]: 已按照建议修改：

1. `_check_beam/column` 通过 `_has_spacing_metadata()` 守卫，无配筋元数据时跳过 `钢筋净距`，不再误报通过
2. `get_rules()` 已补充 `钢筋净距` 到梁和柱的检查列表

**Review by qinsz01 — APPROVED** [review]
